docs/ENVIRONMENTS.md

```diff
@@ -0,0 +1,99 @@
+# Environments support 
+
+## Environment.ts
+
+In the `src/environments/model.ts` is provided the Environment interface as follows
+
+```
+...
+export interface Environment {
+	production: boolean;
+	hmr: boolean;
+	...
+```
+
+`production`, which is a boolean, will specify is your application runs under a production environment, 
+while `hmr` will indicate if it runs under a Hot Module Replacement environment. 
+
+This interface will then be defined by default in `src/environments/environment.ts`: 
+```
+...
+export const environment: Environment = {
+	production: false,
+	hmr: false,
+	showDevModule: true,
+	...
+```
+
+This file will then be replaced by the file needed for the environment in `angular.json` config file: 
+
+```
+...
+"configurations": {
+            "hmr": {
+              "fileReplacements": [
+                {
+                  "replace": "src/environments/environment.ts",
+                  "with": "src/environments/environment.hmr.ts"
+                }
+              ]
+            },
+            "production": {
+              "optimization": true,
+              "outputHashing": "all",
+              "sourceMap": false,
+              "extractCss": true,
+              "namedChunks": false,
+              "aot": true,
+              "extractLicenses": true,
+              "vendorChunk": false,
+              "buildOptimizer": true,
+              "fileReplacements": [
+                {
+                  "replace": "src/environments/environment.ts",
+                  "with": "src/environments/environment.prod.ts"
+                }
+              ]
+            },
+            "e2e.prod": {
+              "optimization": true,
+              "outputHashing": "all",
+              "sourceMap": false,
+              "extractCss": true,
+              "namedChunks": false,
+              "aot": true,
+              "extractLicenses": true,
+              "vendorChunk": false,
+              "buildOptimizer": true,
+              "fileReplacements": [
+                {
+                  "replace": "src/environments/environment.ts",
+                  "with": "src/environments/environment.e2e.prod.ts"
+                }
+              ]
+        	 ...
+```
+
+## How to add a new environment? 
+
+You can add as many environments as you want, as long as you define their options and file replacement in `angular.json`:
+and you add the actual needed file in `src/environments`:
```

Comment:
As discussed with @SuperITMan we should provide our own Environment interface which they should extend in case they want to add new environments, otherwise they might change all the interface and break the Stark code because we are relying in that interface